src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -86,8 +85,15 @@ public void Clear()
 		{
 			// 6.0 TODO: inline the call.
 #pragma warning disable 612
-			return Put(key, returnTypes, result, queryParameters.NaturalKeyLookup, session);
+			var cached = Put(key, returnTypes, result, queryParameters.NaturalKeyLookup, session);
```

Comment:
Can't we instead of caching additional value wrap result in DTO class with required data.
Something like `Put(..., new AutDiscoveryCacheValue(result, aliases))`

Reply:
This would be a breaking change for some cache serializers (like the json one), which would need to know about that DTO.

The cached data is an list of objects, mainly arrays of atomic types. So, we may instead handle that in `GetCacheableResult` which already adds the timestamps in that list.

Reply:
> This would be a breaking change for some cache serializers (like the json one), which would need to know about that DTO

But doesn't it mean the same issue applies with your initial implementation? Shouldn't such serializers also know about `QueryAliasesKey`?

Reply:
Yes I was overlooking this trouble for that new key type.

Reply:
Actually no, the key is not serialized, so, no trouble with that `QueryAliasesKey` class, excepted that solution use more code.

Reply:
> Actually no, the key is not serialized,?

How? Or is it just implementation detail of our cache? Something like `ToString()` call? But then it's just no trouble with this particular implementation...

Reply:
Depending on the cache, the key is used "as is", or its string representation is used . (It can be appended with its hashcode.)

Prior to latest version, the CoreDistributedCahe was adding the key into the cached data, which was causing the [JsonSerializer to fail](https://github.com/nhibernate/NHibernate-Caches/issues/106). This has been fixed.